32bit build broken

e.g. i386, armhf
```
[ 85%] Building CXX object source/main/CMakeFiles/RoR.dir/resources/otc_fileformat/OTCFileformat.cpp.o
In file included from /build/rigs-of-rods/parts/ror/build/source/main/cotire/RoR_CXX_prefix.cxx:120:0,
                 from /build/rigs-of-rods/parts/ror/build/source/main/cotire/RoR_CXX_prefix.hxx:4:
/build/rigs-of-rods/stage/usr/include/rapidjson/document.h: In instantiation of ‘typename rapidjson::internal::DisableIf<typename rapidjson::internal::RemoveSfinaeTag<rapidjson::internal::SfinaeTag& (*)(rapidjson::internal::OrExpr<rapidjson::internal::IsPointer<T>, rapidjson::internal::IsGenericValue<T> >)>::Type, typename rapidjson::internal::RemoveSfinaeTag<rapidjson::internal::SfinaeTag& (*)(rapidjson::GenericValue<Encoding, Allocator>&)>::Type>::Type rapidjson::GenericValue<Encoding, Allocator>::AddMember(rapidjson::GenericValue<Encoding, Allocator>&, T, Allocator&) [with T = long int; Encoding = rapidjson::UTF8<>; Allocator = rapidjson::MemoryPoolAllocator<>; typename rapidjson::internal::DisableIf<typename rapidjson::internal::RemoveSfinaeTag<rapidjson::internal::SfinaeTag& (*)(rapidjson::internal::OrExpr<rapidjson::internal::IsPointer<T>, rapidjson::internal::IsGenericValue<T> >)>::Type, typename rapidjson::internal::RemoveSfinaeTag<rapidjson::internal::SfinaeTag& (*)(rapidjson::GenericValue<Encoding, Allocator>&)>::Type>::Type = rapidjson::GenericValue<rapidjson::UTF8<> >&]’:
/build/rigs-of-rods/stage/usr/include/rapidjson/document.h:1416:25:   required from ‘typename rapidjson::internal::DisableIf<typename rapidjson::internal::RemoveSfinaeTag<rapidjson::internal::SfinaeTag& (*)(rapidjson::internal::OrExpr<rapidjson::internal::IsPointer<T>, rapidjson::internal::IsGenericValue<T> >)>::Type, typename rapidjson::internal::RemoveSfinaeTag<rapidjson::internal::SfinaeTag& (*)(rapidjson::GenericValue<Encoding, Allocator>&)>::Type>::Type rapidjson::GenericValue<Encoding, Allocator>::AddMember(rapidjson::GenericValue<Encoding, Allocator>::StringRefType, T, Allocator&) [with T = long int; Encoding = rapidjson::UTF8<>; Allocator = rapidjson::MemoryPoolAllocator<>; typename rapidjson::internal::DisableIf<typename rapidjson::internal::RemoveSfinaeTag<rapidjson::internal::SfinaeTag& (*)(rapidjson::internal::OrExpr<rapidjson::internal::IsPointer<T>, rapidjson::internal::IsGenericValue<T> >)>::Type, typename rapidjson::internal::RemoveSfinaeTag<rapidjson::internal::SfinaeTag& (*)(rapidjson::GenericValue<Encoding, Allocator>&)>::Type>::Type = rapidjson::GenericValue<rapidjson::UTF8<> >&; rapidjson::GenericValue<Encoding, Allocator>::StringRefType = rapidjson::GenericStringRef<char>; typename Encoding::Ch = char]’
/build/rigs-of-rods/parts/ror/src/source/main/resources/CacheSystem.cpp:573:128:   required from here
/build/rigs-of-rods/stage/usr/include/rapidjson/document.h:1345:29: error: call of overloaded ‘GenericValue(long int&)’ is ambiguous
         GenericValue v(value);
                             ^
/build/rigs-of-rods/stage/usr/include/rapidjson/document.h:748:14: note: candidate: rapidjson::GenericValue<Encoding, Allocator>::GenericValue(float) [with Encoding = rapidjson::UTF8<>; Allocator = rapidjson::MemoryPoolAllocator<>]
     explicit GenericValue(float f) RAPIDJSON_NOEXCEPT : data_() { data_.n.d = static_cast<double>(f); data_.f.flags = kNumberDoubleFlag; }
              ^
/build/rigs-of-rods/stage/usr/include/rapidjson/document.h:745:14: note: candidate: rapidjson::GenericValue<Encoding, Allocator>::GenericValue(double) [with Encoding = rapidjson::UTF8<>; Allocator = rapidjson::MemoryPoolAllocator<>]
     explicit GenericValue(double d) RAPIDJSON_NOEXCEPT : data_() { data_.n.d = d; data_.f.flags = kNumberDoubleFlag; }
              ^
/build/rigs-of-rods/stage/usr/include/rapidjson/document.h:733:14: note: candidate: rapidjson::GenericValue<Encoding, Allocator>::GenericValue(uint64_t) [with Encoding = rapidjson::UTF8<>; Allocator = rapidjson::MemoryPoolAllocator<>; uint64_t = long long unsigned int]
     explicit GenericValue(uint64_t u64) RAPIDJSON_NOEXCEPT : data_() {
              ^
/build/rigs-of-rods/stage/usr/include/rapidjson/document.h:718:14: note: candidate: rapidjson::GenericValue<Encoding, Allocator>::GenericValue(int64_t) [with Encoding = rapidjson::UTF8<>; Allocator = rapidjson::MemoryPoolAllocator<>; int64_t = long long int]
     explicit GenericValue(int64_t i64) RAPIDJSON_NOEXCEPT : data_() {
              ^
/build/rigs-of-rods/stage/usr/include/rapidjson/document.h:712:14: note: candidate: rapidjson::GenericValue<Encoding, Allocator>::GenericValue(unsigned int) [with Encoding = rapidjson::UTF8<>; Allocator = rapidjson::MemoryPoolAllocator<>]
     explicit GenericValue(unsigned u) RAPIDJSON_NOEXCEPT : data_() {
              ^
/build/rigs-of-rods/stage/usr/include/rapidjson/document.h:706:14: note: candidate: rapidjson::GenericValue<Encoding, Allocator>::GenericValue(int) [with Encoding = rapidjson::UTF8<>; Allocator = rapidjson::MemoryPoolAllocator<>]
     explicit GenericValue(int i) RAPIDJSON_NOEXCEPT : data_() {
              ^
/build/rigs-of-rods/stage/usr/include/rapidjson/document.h:605:5: note: candidate: rapidjson::GenericValue<Encoding, Allocator>::GenericValue(const rapidjson::GenericValue<Encoding, Allocator>&) [with Encoding = rapidjson::UTF8<>; Allocator = rapidjson::MemoryPoolAllocator<>]
     GenericValue(const GenericValue& rhs);
     ^
/build/rigs-of-rods/stage/usr/include/rapidjson/document.h:598:5: note: candidate: rapidjson::GenericValue<Encoding, Allocator>::GenericValue(rapidjson::GenericValue<Encoding, Allocator>&&) [with Encoding = rapidjson::UTF8<>; Allocator = rapidjson::MemoryPoolAllocator<>]
     GenericValue(GenericValue&& rhs) RAPIDJSON_NOEXCEPT : data_(rhs.data_) {
     ^
```

--- a/source/main/resources/CacheSystem.cpp
+++ b/source/main/resources/CacheSystem.cpp
@@ -484,14 +484,14 @@ void CacheSystem::ExportEntryToJson(rapidjson::Value& j_entries, rapidjson::Docu
 
     // Common details
     j_entry.AddMember("usagecounter",         entry.usagecounter,                                          j_doc.GetAllocator());
-    j_entry.AddMember("addtimestamp",         entry.addtimestamp,                                          j_doc.GetAllocator());
+    j_entry.AddMember("addtimestamp",         static_cast<int64_t>(entry.addtimestamp),                    j_doc.GetAllocator());
     j_entry.AddMember("resource_bundle_type", rapidjson::StringRef(entry.resource_bundle_type.c_str()),    j_doc.GetAllocator());
     j_entry.AddMember("resource_bundle_path", rapidjson::StringRef(entry.resource_bundle_path.c_str()),    j_doc.GetAllocator());
     j_entry.AddMember("fpath",                rapidjson::StringRef(entry.fpath.c_str()),                   j_doc.GetAllocator());
     j_entry.AddMember("fname",                rapidjson::StringRef(entry.fname.c_str()),                   j_doc.GetAllocator());
     j_entry.AddMember("fname_without_uid",    rapidjson::StringRef(entry.fname_without_uid.c_str()),       j_doc.GetAllocator());
     j_entry.AddMember("fext",                 rapidjson::StringRef(entry.fext.c_str()),                    j_doc.GetAllocator());
-    j_entry.AddMember("filetime",             entry.filetime,                                              j_doc.GetAllocator()); 
+    j_entry.AddMember("filetime",             static_cast<int64_t>(entry.filetime),                        j_doc.GetAllocator()); 
     j_entry.AddMember("dname",                rapidjson::StringRef(entry.dname.c_str()),                   j_doc.GetAllocator());
     j_entry.AddMember("categoryid",           entry.categoryid,                                            j_doc.GetAllocator());
     j_entry.AddMember("uniqueid",             rapidjson::StringRef(entry.uniqueid.c_str()),                j_doc.GetAllocator());

